src/test/kotlin/com/metalbear/mirrord/MirrordPluginTest.kt

```diff
@@ -0,0 +1,213 @@
+                ),
+                emptyList(),
+                listOf(ideDownloader.downloadRobotPlugin(tmpDir), pluginPath),
+                tmpDir
+            )
+            println("waiting for IDE...")
+            waitForIgnoringError(ofMinutes(3)) { remoteRobot.callJs("true") }
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun cleanUp() {
+            val xpathPage = URL("http://localhost:8082/").readText()
+            // create a file in build/reports
+            val file = File("build/reports/robot-page.html")
+            file.writeText(xpathPage)
+            ideaProcess?.destroy()
+            tmpDir.toFile().deleteRecursively()
+        }
+    }
+
+    @Test
+    @Video
+    fun testMirrordFlow() = with(remoteRobot) {
+        step("Welcome Frame") {
+            welcomeFrame {
+                steps?.openProject(System.getProperty("test.workspace"))
+            }
+        }
+        idea {
+            step("Open `app.py`") {
+                // sometimes the projectViewTree does not open, this is quite random
+                // as a workaround we try to open it again
+                var clickToOpenProject = false
+                val projectViewTree = try {
+                    projectViewTree
+                } catch (e: Exception) {
+                    leftStripe {
+                        findText("Project").click()
+                    }
+                    clickToOpenProject = true
+                    projectViewTree
+                }
+                with(projectViewTree) {
+                    if (clickToOpenProject) {
+                        waitFor(ofSeconds(30)) {
+                            hasText("test-workspace")
+                        }
+                        findText("test-workspace").doubleClick()
+                    }
+                    waitFor(ofSeconds(30)) {
+                        hasText("app.py")
+                    }
+                    findText("app.py").doubleClick()
+                    editorTabs {
+                        waitFor {
+                            isFileOpened("app.py")
+                        }
+                    }
+                }
+                step("Set up Poetry Environment") {
+                    // blue hover appears on top of the text window asking
+                    // to set up poetry environment
+                    fileIntention {
+                        val setUpPoetry = setUpPoetry
+                        setUpPoetry.click()
+                        waitFor {
+                            !setUpPoetry.isShowing
+                        }
+                    }
+                    statusBar {
+                        // need to make sure poetry is not doing anything
+                        try {
+                            waitFor {
+                                !poetryProgress.isShowing
+                            }
+                        } catch (e: Exception) {
+                            // ignore
```

Comment:
Why are you ignoring this?

Reply:
so the issue is we dont want to do anything if intellij is doing anything with poetry setup, same as indexing so i try to check if its happening then just ignore if it times out that means nothing is in the status bar related to poetry

Reply:
But your waitFor clause is waiting for poetryProgress *not* to show, so timeout means that poetryProgress is still showing, right?

Reply:
yes correct, i think i should remove the try catch

Reply:
thanks i have removed it